ykrt/src/mt.rs

```diff
@@ -949,4 +957,41 @@ mod tests {
         mtt.control_point(loc, simple_interp_step, &mut ctx);
         assert_eq!(ctx.run, 2);
     }
+
+    #[test]
+    fn loop_interpreter() {
+        use std::{thread, time};
+        let mut mtt = MTBuilder::new().hot_threshold(2).init();
+
+        let loc = Location::new();
+
+        struct InterpCtx {
+            counter: u8,
+        }
+
+        #[interp_step]
+        fn simple_interp_step(ctx: &mut InterpCtx) -> bool {
+            let a = ctx.counter;
+            if ctx.counter < 100 {
+                ctx.counter = a + 1;
+            } else {
+                ctx.counter = a + 2;
+            }
+            if ctx.counter > 110 {
+                return true; // Exit the interpreter.
+            }
+            false
+        }
+
+        let mut ctx = InterpCtx { counter: 0 };
+        loop {
+            if mtt.control_point(Some(&loc), simple_interp_step, &mut ctx) {
+                break;
+            }
+            // Slow down the interpreter so we get a chance to compile stuff.
+            thread::sleep(time::Duration::from_millis(10));
```

Comment:
Sleeping isn't going to work in general (all we need is a faster computer...). The only way to do this is to copy other tests which check what the state of the `Location` is.

Reply:
Okay, will fix.

Reply:
Fixed in 7e160bbd13c050265ff1e5a8292f471e209dffd6.